src/docs/sphinx/theory_reference/solid.rst

```diff
@@ -0,0 +1,199 @@
+   \end{align*}
+
+where :math:`\nabla_X` is the gradient with respect to the reference (material) coordinates.
+
+Material Models
+===============
+
+Serac uses *hyperelastic* material formulations, i.e. materials that behave elastically under large deformations. Mathematically,
+this implies they are derived from a *strain energy density* function :math:`W=W(\mathbf{F})`. It can be shown that
+
+.. math::
+
+   \sigma(\mathbf{F}) = \frac{1}{\text{det}\mathbf{F}} \frac{\partial W}{\partial \mathbf{F}} \mathbf{F}^T  = \frac{1}{\text{det}\mathbf{F}} \mathbf{P}  \mathbf{F}^T
+
+where
+
+.. math::
+
+   \mathbf{P} =  \frac{\partial W}{\partial \mathbf{F}} = {\text{det}}\mathbf{F} \sigma \mathbf{F}^{-T}
+
+is the *first Piola-Kirchhoff stress*. Serac currently only has two material models. First, a neo-Hookean material where
+   
+.. math::
+
+   \begin{align*}
+   W(\mathbf{F}) &= \frac{\mu}{2}(\bar{I}_1 - \text{dim}) + \frac{K}{2}(\text{det}\mathbf{F} - 1)^2 \\
+   \bar{I}_1 &= \frac{\text{trace}(\mathbf{F}\mathbf{F}^T)}{(\text{det}\mathbf{F})^{2/\text{dim}}}
+   \end{align*}
+
+and :math:`\mu` and :math:`K` are the shear and bulk modulus, respectively. This definition also
+implies that the 2D simulations are using a plane strain assumption. The second model is a small strain isotropic linear elastic material where
+
+.. math::
+
+   \begin{align*}
+   \sigma(\epsilon) &= \lambda \text{trace}(\epsilon) \mathbf{I} + 2\mu \epsilon \\
+   \epsilon &= \frac{1}{2}\left(\mathbf{F} + \mathbf{F}^T \right) - \mathbf{I} \\
+   \lambda &= K - \frac{2}{\text{dim}} \mu
+   \end{align*}
+
+and :math:`\epsilon` is the linearized strain tensor. Note that this model is only valid for small strains 
+where the neo-Hookean model is nearly equivalent. It is included mostly for testing purposes.
+
+Optionally, we can also model linear viscoelasticity by adding
+
+.. math::
+
+   \sigma(\dot{\epsilon}) = \eta \dot{\epsilon}
+
+to the stress calculations in dynamic simulations.
+
+Discretization
+==============
+
+We discretize the displacement field using nodal shape functions, i.e.
+
+.. math::
+
+   \mathbf{u} (\mathbf{X}) = \sum_{a=1}^n N^a (\mathbf{X}) \mathbf{u}^a
+
+where :math:`\mathbf{u}^a` are the degrees of freedom. We can then calculate the deformation gradient by
+
+.. math::
+
+   \mathbf{F} = \mathbf{I} + \sum_{a=1}^n \frac{\partial N^a}{\partial \mathbf{X}} \mathbf{u}^a
+
+and substitute these quantities back into the weak form to obtain the vector-valued discrete residual equation
+
+.. math::
+
+   \int_{\Omega_0} \sigma \frac{\partial N^a}{\partial \mathbf{X}} \mathbf{F}^{-1} \text{det}\mathbf{F} \, dV_0  - \int_{\Omega_0} \rho_0 \mathbf{b} N^a dV_0  - \int_{\Gamma_{N_0}} \mathbf{t}^* N^a \, dA_0 + \int_{\Omega_0} \rho_0\ddot{\mathbf{u}}N^a\, dV_0 = 0
+
+where :math:`\mathbf{t}^*` is the traction applied in the reference configuration. 
```

Comment:
Do you really apply the traction in the reference configuration?  That wasn't what I remembered from the code.

It might make sense, as you are using the Cauchy Stress here, to simply write everything in the current configuration, or alternatively put both formulations here.

Reply:
Yes - the MFEM integrators actually act on the undeformed (reference) mesh. The `compute_on_reference` flag is what optionally transforms the input traction to the current configuration using Nanson's rule.